str.go

```diff
@@ -0,0 +1,91 @@
+package merkletree
```

Comment:
Same as with the history, I would put this in a separate package together with the history